### PR TITLE
Notes about implementing Ethernet-over-USB

### PR DESCRIPTION
**Pull request by arturkow2000**

### CODE REVIEW

File: dev-notes/implementing-eth-over-usb.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
+# Implementing Ethernet-over-USB
+
+This document describes various stuff regarding implementation of
+Ethernet-over-USB. Before commencing implementation we have conducted research
+(see [eth-over-usb-research.md](eth-over-usb-research.md)) for more information.
+To sum up, thats what we have decided:
```
Comment (0xDen) [inline dev-notes/implementing-eth-over-usb.md:6]: ...that`'`s...

File: dev-notes/implementing-eth-over-usb.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+  Managers's bug, but it affects Fobnail users. You can workaround this by
+  appending following lines to `/etc/NetworkManager/NetworkManager.conf`
+  ```ini
+  [keyfile]
+  unmanaged-devices=interface-name:usb0
+  ```
+
+## Testing
+
+If you environment is properly set up, app can be flashed by simply running
```
Comment (0xDen) [inline dev-notes/implementing-eth-over-usb.md:96]: If `your` environment...

File: dev-notes/implementing-eth-over-usb.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+```shell
+$ cargo embed
+```
+
+`cargo embed` spawn RTT console used for debugging. On host you should see new
```
Comment (0xDen) [inline dev-notes/implementing-eth-over-usb.md:102]: ...spawn`s`?

File: dev-notes/implementing-eth-over-usb.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
+```
+
+Assign IP address to `usb0`, Fobnail token is at `169.254.0.1`
+
+```shell
+$ ifconfig usb0 169.254.0.2
+```
+
+Fobnail should start printing debug information on RTT console. Note that these
```
Comment (0xDen) [inline dev-notes/implementing-eth-over-usb.md:121]: should start `to print`...?

**Review by 0xDen — APPROVED** [review]
Small remarks

**Comment by macpijan** [comment]
@arturkow2000 Please address the comments. Please also move this document (and also the one with research) from `dev-notes` to `docs` directory. Those are great documents presenting the progress of our work. We would like them to be available in the fobnail.3mdeb.com.

**Review by macpijan — APPROVED** [review]

**Pull request merged**